lib/bolt/target.rb

```diff
@@ -21,7 +21,34 @@ def self.parse_urls(urls)
     def initialize(uri, options = nil)
       @uri = uri
       @uri_obj = parse(uri)
+      @protocol = @uri_obj.scheme
+      @user = @uri_obj.user
+      @password = @uri_obj.password
+      @port = @uri_obj.port
       @options = options || {}
+      @configured = false
+    end
+
+    def update_conf(conf)
+      @protocol = conf[:transport] unless protocol
+
+      t_conf = conf[:transports][protocol.to_sym]
```

Comment:
It might be clearer to store a `@transport_conf` instance variable and write accessors that handle picking between the conf value and the assigned value. An `options` method could handle removing the url keys when requested, or we could continue to store those in their own variable. That would also simplify the `invalidate_conf` bit (though if it's actually not used, maybe that's a smaller win), since it could just unassign the `@transport_conf`.

Reply:
I had something closer to that originally. I'll change it back.

invalidate_conf was intended not to actually remove the conf but to mark is as an 'expired configuration cache' so the target would get it's configuration updated. It's not used and should just be removed.